### PR TITLE
feat: Caffeine based Api-Key caching

### PR DESCRIPTION
**Pull request by ricardas-buc**
Refs: XRDDEV-444

### CODE REVIEW

**Comment by ricardas-buc** [comment]
[rerun-ci]

File: doc/Manuals/ug-cs_x-road_6_central_server_user_guide.md [hunk 1]
```diff
@@ -266,6 +268,18 @@ In order to check the status of the nodes in an HA setup, execute the following
 
 `curl -k https://localhost:4000/public_system_status/check_ha_cluster_status`
 
+## 3.3 API key considerations in High-Availability setup
+
+API keys are cached in memory, which is typically not a problem in non-clustered central server configuration.
```
Comment (raits) [inline doc/Manuals/ug-cs_x-road_6_central_server_user_guide.md:273]: Let's write central server as Central Server.

File: doc/Manuals/ug-ss_x-road_6_security_server_user_guide.md [hunk 2]
```diff
@@ -2698,17 +2699,16 @@ curl -X DELETE -u <user>:<password> https://localhost:4000/api/v1/api-keys/60  -
...
 
-For example, you may revoke an API key from node 1 but node 2 is not aware of this, and still grants access to
-REST API endpoints with this API key.
+For instance, revoking an API key from `node 1` may not be recognized by `node 2`, which can still grant access to REST API endpoints with the revoked API key. To address this issue, there are a few potential solutions:
 
-If you operate such a configuration, you need to target all REST API operations to the same security server node,
-or otherwise ensure that caching will not create problems (for example, always restart REST API modules when API key
-operations are executed).
+- **Option A:** Consider decreasing [time-to-live](ug-syspar_x-road_v6_system_parameters.md#39-management-rest-api-parameters-proxy-ui-api) value for API key cache from the default of **60 seconds** to a more lenient value. Doing so will reduce the risk of stale values being returned, thus improving security.
+- **Option B:** Direct all REST API operations to the same security server node.
```
Comment (raits) [inline doc/Manuals/ug-ss_x-road_6_security_server_user_guide.md:2709]: Let's change security server to Security Server here similar to what we do with the Central Server.

File: doc/Manuals/ug-cs_x-road_6_central_server_user_guide.md [hunk 3]
```diff
@@ -266,6 +268,18 @@ In order to check the status of the nodes in an HA setup, execute the following
...
+However, in case of High-Availability setup, the caches of different nodes can become out of sync.
+
+For instance, revoking an API key from `node 1` may not be recognized by `node 2`, which can still grant access to REST API endpoints with the revoked API key. To address this issue, there are a few potential solutions:
+
+- **Option A:** Consider decreasing [time-to-live](ug-syspar_x-road_v6_system_parameters.md#413-center-parameters-admin-service) value for API key cache from the default of **60 seconds** to a more lenient value. Doing so will reduce the risk of stale values being returned, thus improving security.
+- **Option B:** Direct all REST API operations to the same security server node.
```
Comment (raits) [inline doc/Manuals/ug-cs_x-road_6_central_server_user_guide.md:279]: This should probably be Central Server instead of Security Server.

**Review by raits — CHANGES_REQUESTED** [review]
A few minor changes, but otherwise, the documentation looks good to me.

**Review by raits — APPROVED** [review]
Documentation looks good to me.

**Review by ovidijusnortal — APPROVED** [review]

**Pull request merged**